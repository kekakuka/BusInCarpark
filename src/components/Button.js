import React from 'react';
export default function Button({ onClick, text, start, disabled }) {
  return (
    <button
      style={{
        background: disabled
          ? 'linear-gradient(45deg,rgba(140,140,140, 0.9) 30%, rgba(180,180,180, 0.9) 90%)'
          : start
          ? 'linear-gradient(45deg,rgba(140,20,40, 0.9) 30%, rgba(100,32,59, 0.9) 90%)'
          : 'linear-gradient(45deg,rgba(40,30,40, 0.9) 30%, rgba(80,72,109, 0.9) 90%)',
        border: 0,
        borderRadius: 3,
        boxShadow: '1px 2px 2px 2px rgba(155, 155, 155, .5)',
        color: '#eee',
        height: 30,
        fontSize: 15,
        padding: '0 10px',
        marginRight: 20,
        width: 90,
        marginBottom: 6
      }}
      disabled={disabled}
      onClick={onClick}
    >
      {text}
    </button>
  );
}
