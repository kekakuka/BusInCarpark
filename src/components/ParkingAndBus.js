import React from 'react';
export default React.memo(function ParkingAndBus({ position: { x = 0, y = -1, f = 0 } }) {
  return (
    <div
      style={{
        width: 300,
        height: 300,
        background: 'url(' + require('../static/images/Parking.png') + ')',
        position: 'relative',
        boxShadow: '4px 4px 10px 3px  rgba(155, 155, 155, .5)'
      }}
    >
      <div
        style={{
          width: 60,
          height: 60,
          position: 'absolute',
          background: 'url(' + require('../static/images/Bus.png') + ')',
          left: x * 60,
          bottom: y * 60,
          textAlign: 'center',
          transform: `rotate(${f * 90}deg)`
        }}
      >
        {y === -1 && 'Out Of Parking'}
      </div>
    </div>
  );
});
