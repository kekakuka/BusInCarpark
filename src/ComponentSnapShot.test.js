import React from 'react';

import renderer from 'react-test-renderer';
import ParkingAndBus from './components/ParkingAndBus';
import Button from './components/Button';

test('ParkingAndBus renders correctly', () => {
  const tree = renderer.create(<ParkingAndBus position={{}} />).toJSON();
  expect(tree.type).toBe('div');
  expect(tree.children[0].children[0]).toBe('Out Of Parking');
  expect(tree).toMatchSnapshot();
});

test('Button renders correctly', () => {
  const tree = renderer.create(<Button onClick={() => {}} text={'A Button'} />).toJSON();
  expect(tree.type).toBe('button');
  expect(tree.children).toEqual(['A Button']);
  expect(tree).toMatchSnapshot();
});
