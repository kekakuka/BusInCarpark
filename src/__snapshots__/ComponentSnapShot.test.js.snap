// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "background": "linear-gradient(45deg,rgba(40,30,40, 0.9) 30%, rgba(80,72,109, 0.9) 90%)",
      "border": 0,
      "borderRadius": 3,
      "boxShadow": "1px 2px 2px 2px rgba(155, 155, 155, .5)",
      "color": "#eee",
      "fontSize": 15,
      "height": 30,
      "marginBottom": 6,
      "marginRight": 20,
      "padding": "0 10px",
      "width": 90,
    }
  }
>
  A Button
</button>
`;

exports[`ParkingAndBus renders correctly 1`] = `
<div
  style={
    Object {
      "background": "url(Parking.png)",
      "boxShadow": "4px 4px 10px 3px  rgba(155, 155, 155, .5)",
      "height": 300,
      "position": "relative",
      "width": 300,
    }
  }
>
  <div
    style={
      Object {
        "background": "url(Bus.png)",
        "bottom": -60,
        "height": 60,
        "left": 0,
        "position": "absolute",
        "textAlign": "center",
        "transform": "rotate(0deg)",
        "width": 60,
      }
    }
  >
    Out Of Parking
  </div>
</div>
`;
